ampersand-state.js

```diff
@@ -177,11 +178,12 @@ assign(Base.prototype, Events, {
             }
 
             isEqual = this._getCompareForType(def.type);
+            onChange = this._getOnChangeForType(def.type);
             dataType = this._dataTypes[def.type];
 
             // check type if we have one
             if (dataType && dataType.set) {
-                cast = dataType.set(newVal);
+                cast = dataType.set.call(this, newVal, options);
```

Comment:
+1 on unbinding, would be a breaking change for those using compare to do
stuff currently

On Thu, Oct 22, 2015 at 9:36 AM, chesles <notifications@github.com> wrote:

> In ampersand-state.js
> <https://github.com/AmpersandJS/ampersand-state/pull/202#discussion_r42755662>
> :
>
> >              dataType = this._dataTypes[def.type];
> >
> >              // check type if we have one
> >              if (dataType && dataType.set) {
> > -                cast = dataType.set(newVal);
> > +                cast = dataType.set.call(this, newVal, options);
>
> Agreed, good point @latentflip <https://github.com/latentflip>. Making
> this available in set could enable problems like the ones fixed in this
> PR (side-effects where there shouldn't be). To that point, maybe the
> compare function should be unbound from this as well and encourage use of
> onChange?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AmpersandJS/ampersand-state/pull/202/files#r42755662>.
>